pyodide_build/config.py

```diff
@@ -122,25 +122,81 @@ def _get_make_environment_vars(self) -> Mapping[str, str]:
         Load environment variables from Makefile.envs
         """
         environment = {}
-        result = run_command(
-            ["make", "-f", str(self.pyodide_root / "Makefile.envs"), ".output_vars"],
-            env={"PYODIDE_ROOT": str(self.pyodide_root)},
-            err_msg="ERROR: Failed to load environment variables from Makefile.envs",
-        )
-
-        for line in result.stdout.splitlines():
-            equalPos = line.find("=")
-            if equalPos != -1:
-                varname = line[0:equalPos]
-
-                if varname not in BUILD_VAR_TO_KEY:
-                    continue
+        env = os.environ | {"PYODIDE_ROOT": str(self.pyodide_root)}
+        makefile_path = self.pyodide_root / "Makefile.envs"
+
+        # Try using make first if available
+        try:
+            result = run_command(
+                ["make", "-f", str(makefile_path), ".output_vars"],
+                env=env,
+                err_msg="ERROR: Failed to load environment variables from Makefile.envs",
+            )
```

Comment:
This try/catch is broader than it needs to be. How about:
```
    result = None
    try:
        result = run_command(...)
    except CommandFailedException: # Not sure what we raise
        pass
    if result is None:
        logger.debug(
            "make command not available, using fallback Makefile.envs parser"
        )
        return _parse_makefile_envs(env=env, makefile_path=makefile_path)
```

Reply:
Or what about checking if the platform is windows and avoiding using a try/catch at all?

Reply:
Makes sense.

> Or what about checking if the platform is windows and avoiding using a try/catch at all?

Changed to this way.